src/main/java/az/gdg/msauth/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+package az.gdg.msauth.service.impl;
+
+import az.gdg.msauth.dao.UserRepository;
+import az.gdg.msauth.mapper.UserMapper;
+import az.gdg.msauth.model.dto.MailDTO;
+import az.gdg.msauth.model.dto.ResetPasswordDTO;
+import az.gdg.msauth.model.dto.UserDTO;
+import az.gdg.msauth.model.entity.UserEntity;
+import az.gdg.msauth.exception.WrongDataException;
+import az.gdg.msauth.security.model.Status;
+import az.gdg.msauth.security.model.dto.UserInfo;
+import az.gdg.msauth.security.exception.AuthenticationException;
+import az.gdg.msauth.security.model.Role;
+import az.gdg.msauth.model.dto.UserInfoForBlogService;
+import az.gdg.msauth.security.service.AuthenticationService;
+import az.gdg.msauth.security.util.TokenUtil;
+import az.gdg.msauth.service.EmailService;
+import az.gdg.msauth.service.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.UUID;
+
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final TokenUtil tokenUtil;
+    private final AuthenticationService authenticationService;
+    private final EmailService emailService;
+    private static final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
+
+    public UserServiceImpl(UserRepository userRepository, AuthenticationService authenticationService,
+                           EmailService emailService, TokenUtil tokenUtil) {
+        this.userRepository = userRepository;
+        this.authenticationService = authenticationService;
+        this.emailService = emailService;
+        this.tokenUtil = tokenUtil;
+    }
+
+    public void signUp(UserDTO userDTO) {
+        logger.info("ActionLog.Sign up user.Start");
+
+        UserEntity checkedEmail = userRepository.findByEmail(userDTO.getEmail());
+        if (checkedEmail != null) {
+            logger.error("ActionLog.WrongDataException.Thrown");
+            throw new WrongDataException("This email already exists");
+        }
+
+        String password = new BCryptPasswordEncoder().encode(userDTO.getPassword());
+        String code = UUID.randomUUID().toString();
+        UserEntity userEntity = UserEntity
+                .builder()
+                .name(userDTO.getName())
+                .surname(userDTO.getSurname())
+                .username(userDTO.getEmail())
+                .email(userDTO.getEmail())
+                .password(password)
+                .accountVerificationCode(code)
```

Comment:
Why you need accountVerificationCode, we discussed that we will use token in url